src/api/external/ami.py

```diff
@@ -1,26 +1,27 @@
-import json
-import os
-
-import requests
-
-AMI_ID = 'https://aviatrix-download.s3-us-west-2.amazonaws.com/AMI_ID/ami_id.json'
-DEV_FLAG = "dev_flag"
-
-
-def check_ami_id(ami_id):
-    """ Check if AMI is latest"""
-    if os.path.exists(DEV_FLAG):
-        print("Skip checking AMI ID for dev work")
-        return True
-    print("Verifying AMI ID")
-    resp = requests.get(AMI_ID)
-    ami_dict = json.loads(resp.content)
-    for image_type in ami_dict:
-        if ami_id in list(ami_dict[image_type].values()):
-            print("AMI is valid")
-            return True
-    print("AMI is not latest. Cannot enable Controller HA. Please backup restore to the latest AMI"
-          "before enabling controller HA")
-    return False
-
-
+import json
+import os
+
+import requests
+
+from common.constants import DEV_FLAG
+
+AMI_ID = 'https://aviatrix-download.s3-us-west-2.amazonaws.com/AMI_ID/ami_id.json'
+
+
+def check_ami_id(ami_id):
+    """ Check if AMI is latest"""
+    if os.path.exists(DEV_FLAG):
+        print("Skip checking AMI ID for dev work")
+        return True
+    print("Verifying AMI ID")
+    resp = requests.get(AMI_ID)
+    ami_dict = json.loads(resp.content)
+    for image_type in ami_dict:
+        if ami_id in list(ami_dict[image_type].values()):
+            print("AMI is valid")
+            return True
```

Comment:
optional:

```
try:
    resp = requests.get(AMI_ID)
    resp.raise_for_status()  # Raises an HTTPError if the response was an error
    ami_dict = resp.json()  # Directly parse the JSON response

    if any(ami_id in values for values in ami_dict.values()):
        print("AMI is valid")
        return True
    else:
          print("AMI is not latest. Cannot enable Controller HA. Please backup restore to the latest AMI"
          "before enabling controller HA")
except requests.RequestException as e:
    print(f"Error checking AMI ID: {e}")
    return False
```

Reply:
This is a little different from the original; `ami_dict` is a nested dictionary. I added the HTTP error checking and added a unit test